lib/Zonemaster/LDNS/IDN.pm

```diff
@@ -0,0 +1,47 @@
+package Zonemaster::LDNS::IDN;
+
+use strict;
+use warnings;
+
+use Carp;
+
+use parent 'Exporter';
+our @EXPORT_OK = qw[has_idn to_idn];
+
+my $idn_available = 0;
+
+eval {
+    use Net::LibIDN2 ':all';
+};
+
+if ( ! $@ ) {
+    $idn_available = 1;
+}
+
+sub has_idn {
+    return $idn_available;
+}
+
+sub to_idn {
+    if ( !has_idn() ) {
+        croak( "Module Net::LibIDN2 not installed." );
+    }
+
+    my @dst;
+    for ( @_ ) {
+        my $rc = -1;
+        my $out = Net::LibIDN2::idn2_to_ascii_8( $_, IDN2_NFC_INPUT, $rc );
```

Comment:
We have to figure out what flags to set. I our context U-label to A-label conversion should rather be "raw" by default than having some magic. If we use magic, and that could make sense in some contexts, then that should be explicit, not hidden. Making input NFC is magic.

Reply:
The module says:

> _Note that the input string must be encoded in UTF-8 and be in Unicode NFC form._
https://metacpan.org/pod/Net::LibIDN2#Net::LibIDN2::idn2_lookup_u8($src-[,-$flags-[,-$rc]]);

I don't know enough on Unicode and normalization forms to properly know if the string in `$_` is in a proper form.

@matsduf do you have any idea on which flags to use? Or are we facing a limitation that we can't lift and would prevent us to use this Perl module (we could still go with the proposal in #133)?

Reply:
I do not think that there are any limitations that prevent us from using it. It is OK if we let it do some magic as long as we document that so that we do not get surprised later. I will make a suggestions.